plaid/src/bin/config_check.rs

```diff
@@ -0,0 +1,52 @@
+use clap::{Arg, Command};
+use serde_json::Value;
+use sha3::{Digest, Sha3_256};
+
+fn main() {
+    let matches = Command::new("Config Check")
+        .version(env!("CARGO_PKG_VERSION"))
+        .about("Confirm that secrets interpolated config is what you expect it to be")
+        .arg(
+            Arg::new("config")
+                .help("Path to the configuration toml file")
+                .long("config")
+                .default_value("./plaid/resources/plaid.toml"),
+        )
+        .arg(
+            Arg::new("secrets")
+                .help("Path to the secrets json file")
+                .long("secrets")
+                .default_value("./plaid/private-resources/secrets.json"),
+        )
+        .get_matches();
+
+    // Read the configuration file
+    let mut config = std::fs::read_to_string(matches.get_one::<String>("config").unwrap())
+        .expect("Failed to read configuration file");
+
+    // Read the secrets file and parse into a serde object
+    let secrets = std::fs::read(matches.get_one::<String>("secrets").unwrap())
+        .expect("Failed to read secrets file");
+
+    let secret_map = serde_json::from_slice::<Value>(&secrets)
+        .unwrap()
+        .as_object()
+        .cloned()
+        .unwrap();
+
+    // Iterate over the secrets we just parsed and replace matching keys in the config
+    for (secret, value) in secret_map {
```

Comment:
Can we abstract this into a function call with shared code in the Plaid library itself? That was as we update that code, it will apply to config check as well>